pkg/validation/ingestion_policies_test.go

```diff
@@ -284,3 +288,203 @@ func TestPolicyStreamMapping_ApplyDefaultPolicyStreamMappings_Validation(t *test
 	// Verify the result is valid
 	require.NoError(t, existing.Validate())
 }
+
+func Test_PolicyStreamMapping_PolicyFor_WithHeaderOverride(t *testing.T) {
+	mapping := PolicyStreamMapping{
+		"policy1": []*PriorityStream{
+			{
+				Selector: `{foo="bar"}`,
+				Priority: 2,
+				Matchers: []*labels.Matcher{
+					labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+				},
+			},
+		},
+		"policy2": []*PriorityStream{
+			{
+				Selector: `{env="prod"}`,
+				Priority: 1,
+				Matchers: []*labels.Matcher{
+					labels.MustNewMatcher(labels.MatchEqual, "env", "prod"),
+				},
+			},
+		},
+	}
+
+	require.NoError(t, mapping.Validate())
+
+	t.Run("without header context, uses normal mapping", func(t *testing.T) {
+		ctx := context.Background()
```

Comment:
Tests now have contexts

```suggestion
		ctx := t.Context()
```